### PR TITLE
Fix incorrect hydration when pooling is disabled

### PR DESCRIPTION
**Pull request by tonivdv**
Fixes issue #980

### CODE REVIEW

File: runtime/lib/formatter/PropelObjectFormatter.php [hunk 1]
```diff
@@ -39,16 +39,21 @@ public function format(PDOStatement $stmt)
             $pks = array();
             $objectsByPks = array();
             while ($row = $stmt->fetch(PDO::FETCH_NUM)) {
-                $object = $this->getAllObjectsFromRow($row);
-                $pk = $object->getPrimaryKey();
 
                 if (false === Propel::isInstancePoolingEnabled()) {
+
+                    list($object, $col) = call_user_func(array($this->peer, 'populateObject'), $row);
```
Comment (staabm) [inline runtime/lib/formatter/PropelObjectFormatter.php]: couldn't this just be
`list($object, $col) = $this->peer->populateObject($row);`
?

**Comment by staabm** [comment]
could you also add a unit test supporting your change?

File: runtime/lib/formatter/PropelObjectFormatter.php (hunk 1 above)
Comment (tonivdv) [inline runtime/lib/formatter/PropelObjectFormatter.php]: Hey @staabm ,

I actually copied this line from the method "getAllObjectsFromRow". Didn't want to change anything else as I don't completely understand yet the framework.

**Comment by tonivdv** [comment]
I wanted to already see what the tests gave on travisci, before setting everything up on my side. If it's green I can add a test.

File: runtime/lib/formatter/PropelObjectFormatter.php (hunk 1 above)
Comment (staabm) [inline runtime/lib/formatter/PropelObjectFormatter.php]: agree.

File: runtime/lib/formatter/PropelObjectFormatter.php [hunk 2]
```diff
@@ -39,16 +39,21 @@ public function format(PDOStatement $stmt)
...
+                    $pk = $object->getPrimaryKey();
+
                     if (isset($objectsByPks[$pk])) {
                         $this->mainObject = $objectsByPks[$pk];
-                        $object = $this->getAllObjectsFromRow($row);
                     }
+                    $object = $this->getAllObjectsFromRow($row);
```
Comment (staabm) [inline runtime/lib/formatter/PropelObjectFormatter.php]: let me sum up your PR:

the actual issue is, that this `$object` does not return the `$pk` you expected on `getPrimaryKey()` call, when instance-pooling is disabled?

Comment (tonivdv) [inline runtime/lib/formatter/PropelObjectFormatter.php]: Not really. The issue is that the method `getAllObjectsFromRow` is called twice when pooling is disabled and is causing issues during hydration when joins (->joinWith() ) are used. What I discovered is, that it needs the primary object to get the primary key in order to set the 'mainObject', but by calling the "getAllObjectsFromRow" it does too much, reason why I extract the first line of that method which only populates the "main object" and not the joined/nested objects. Only calling `getAllObjectsFromRow` once (as when pooling is enabled) solves everything.

Maybe there is a better solution, but I didn't want to make it to fancy as in propel2 it has already be done differently.

**Comment by tonivdv** [comment]
@staabm I got some issues to be able to run the tests locally. If you really need tests I can play with it, but will use this PR to always run the tests. 

In the future having a vagrant box, or docker config for local testing would speed this up.

**Comment by tonivdv** [comment]
@staabm on second thought, debugging through travis is far from ideal. However I can't seem to make the test work locally. 

Are tests in this case really necessary. It doesn't brake current tests?

**Comment by staabm** [comment]
having it covered by unit test would help so we won't regress here in the future.
I dont know this area of propel that good, so I am actually no willing to get my +1 on this.

@marcj what do you think about the PR?

**Comment by marcj** [comment]
Well, this issue is already [fixed in v2](https://github.com/propelorm/Propel2/commit/7a6691c137b233dba1a7ceb4c00b64ffffcad24f). The whole v1 development is currently frozen, due to the focus in v2, so I don't it makes much sense to fix old bugs in legacy code.

**Comment by tonivdv** [comment]
@marcj But on the official web page, you still point to this 'stable' version, right?

**Comment by tbl0605** [comment]
And some people (like me) are still heavily using this version, and would be happy to have it even more stable ;)
Btw, there's a lot of pending patches that could be merged in 'v1' until 'v2' is officially released. So, in all cases, thank you @tonivdv  :)

**Comment by tonivdv** [comment]
@staabm I tried yesterday to have test working correctly with phpstorm etc, but always have issues. Unfortunately I can't spend much more time trying to add tests. You think it's really necessary?

Currently I'm using my fork for the project in order to be able to continue.

**Comment by tonivdv** [comment]
@staabm @marcj I guess we can ignore & close this PR. I'm not using this version anymore either. Plus from old comment this was fixed?!

**Pull request closed without merging**